src/Core/Worker.php

```diff
@@ -300,18 +300,16 @@ public static function execute($queue)
 	 * @param array   $argv        Array of values to be passed to the function
 	 * @param boolean $method_call boolean
 	 * @return void
+	 *
+	 * @throws InternalServerErrorException The current process wasn't properly initialized
```

Comment:
Is this really thrown somewhere? Cannot find it.

Reply:
Hm.. You're right. I thought there was a call to a Process method which can caus an Exception. I've to check it again..

Reply:
`@throws` should be only annotated when the method in question is throwing it, not when a then called method will throw it. It is a minor documentation mistake, nothing big. :-)

Reply:
I'm not sure about that. So lets' say I throw an Exception in a inner method. And the outer method itself doesn't handle any exceptions because it's just a "layer" inbetween, the next outer method will possibly get an exception without knowing it and therefore wouldn't create an adequate Exception handling.

The alternative is that I make a catch in and throw this eception again .. But that's uggly I think..

But for this case, we don't need it :-)

ByTheWay: PHPStorm "cries" when there is a possible returning Exception from a called function without a proper Exception handling or a `@throw` in the phpdoc :)

Reply:
Okay, since PHP doesn't have `throws` keyword for method signatures.

Reply:
OK, now I found it:

`startProcess()` calls `Process::start()` which throws an Exception which is not handled in `startProcess`. So the `@throws` is on the right place because otherwise the caller of `startProcess()` wouldn't get a hint that it may get an Exception when calling this method.